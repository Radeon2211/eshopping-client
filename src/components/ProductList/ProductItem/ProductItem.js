/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import * as SC from './ProductItem.sc';
import { baseURL } from '../../../axios';
import noPhoto from '../../../images/no-photo.png';

const ProductItem = (props) => {
  const {
    data: { name, price, condition, quantitySold, photo, _id },
  } = props;

  let conditionNode = null;
  if (condition !== 'not_applicable') {
    conditionNode = (
      <span className="condition">
        <span className="gray">Condition: </span>
        {condition}
      </span>
    );
  }

  let quantitySoldNode = null;
  if (quantitySold === 1) {
    quantitySoldNode = <span className="quantity-sold gray">1 person bought</span>;
  }
  if (quantitySold > 1) {
    quantitySoldNode = <span className="quantity-sold gray">{quantitySold} people bought</span>;
  }

  return (
    <Link to={`/products/${_id}`}>
      <SC.Wrapper>
        <div className="photo-box">
          <img
            src={photo ? `${baseURL}/products/${_id}/photo` : noPhoto}
            alt="product"
            className="photo"
          />
        </div>
        <div className="data-box">
          <span className="name">{name}</span>
          {conditionNode}
          <span className="price">${price.toFixed(2)}</span>
          {quantitySoldNode}
        </div>
      </SC.Wrapper>
    </Link>
  );
};

ProductItem.propTypes = {
  data: PropTypes.oneOfType([PropTypes.object]).isRequired,
};

export default ProductItem;
