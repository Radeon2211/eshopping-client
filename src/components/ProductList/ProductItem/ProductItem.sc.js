import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const Wrapper = styled.article`
  border-bottom: 1px solid ${({ theme }) => theme.colors.light2};
  display: flex;
  padding: ${({ theme }) => theme.spacings.level3} ${({ theme }) => theme.spacings.level2};
  transition: background-color 0.12s;

  &:hover {
    background-color: rgba(0, 0, 0, 0.07);
    cursor: pointer;
  }

  & .photo-box {
    align-items: center;
    display: flex;
    justify-content: center;
    max-height: 17rem;
    max-width: 13rem;
    margin-right: ${({ theme }) => theme.spacings.level3};
  }

  & .photo {
    height: 100%;
    object-fit: cover;
    width: 100%;
  }

  & .data-box {
    display: flex;
    flex-direction: column;
  }

  & .name {
    font-size: 1.5rem;
  }

  & .condition {
    font-size: 1.3rem;
    text-transform: capitalize;
    margin-top: ${({ theme }) => theme.spacings.level1};
  }

  & .price {
    font-size: 2.4rem;
    margin-top: ${({ theme }) => theme.spacings.level2};
  }

  & .quantity-sold {
    font-size: 1.3rem;
    margin-top: auto;
  }

  & .gray {
    color: ${({ theme }) => theme.colors.light4};
  }
`;
