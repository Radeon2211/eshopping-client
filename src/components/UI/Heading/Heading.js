import React from 'react';
import PropTypes from 'prop-types';
import * as SC from './Heading.sc';

const Heading = React.memo((props) => <SC.Heading {...props}>{props.children}</SC.Heading>);

Heading.defaultProps = {
  align: 'left',
  mgBottom: '',
  mgTop: '',
  lineHeight: '',
  fontSize: '',
};

Heading.propTypes = {
  align: PropTypes.string,
  mgBottom: PropTypes.string,
  mgTop: PropTypes.string,
  lineHeight: PropTypes.string,
  variant: PropTypes.string.isRequired,
  fontSize: PropTypes.string,
  children: PropTypes.node.isRequired,
};

export default Heading;
