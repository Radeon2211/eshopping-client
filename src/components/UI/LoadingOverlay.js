import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Loader from './Loader';

const SC = {};
SC.Wrapper = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1Transparent};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  justify-items: ${({ alignLoader }) => alignLoader};
  left: 0;
  padding-top: ${({ theme }) => theme.spacings.level5};
  position: absolute;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level1};

  & .info {
    font-size: ${({ theme }) => theme.fontSizes.level2};
    margin-top: ${({ theme }) => theme.spacings.level2};
  }

  ${({ zeroPadding }) => {
    if (zeroPadding) {
      return `
        padding: 0;
      `;
    }
    return ``;
  }}
`;

const LoadingOverlay = (props) => {
  const { alignLoader, loaderSize, zeroPadding, disableText } = props;

  const [isInfoVisible, setIsInfoVisible] = useState(false);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsInfoVisible(true);
    }, 8000);
    return () => clearTimeout(timeout);
  }, [setIsInfoVisible]);

  let info = false;
  if (isInfoVisible && !disableText) {
    info = <span className="info">Just a second</span>;
  }

  return (
    <SC.Wrapper alignLoader={alignLoader} zeroPadding={zeroPadding}>
      <Loader size={loaderSize} />
      {info}
    </SC.Wrapper>
  );
};

LoadingOverlay.defaultProps = {
  zeroPadding: false,
  alignLoader: 'center',
  disableText: false,
  loaderSize: '',
};

LoadingOverlay.propTypes = {
  zeroPadding: PropTypes.bool,
  alignLoader: PropTypes.string,
  loaderSize: PropTypes.string,
  disableText: PropTypes.bool,
};

export default LoadingOverlay;
