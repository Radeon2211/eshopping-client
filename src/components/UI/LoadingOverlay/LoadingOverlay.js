import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Loader from '../Loader/Loader';

const SC = {};
SC.Wrapper = styled.div`
  align-items: center;
  background-color: ${({ theme }) => theme.colors.light1Transparent};
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: center;
  left: 0;
  padding-top: ${({ theme }) => theme.spacings.level5};
  position: absolute;
  top: 0;
  width: 100%;
  z-index: ${({ theme }) => theme.zIndexes.level1};

  & .info {
    font-size: ${({ theme }) => theme.fontSizes.level2};
    margin-top: ${({ theme }) => theme.spacings.level2};
  }

  ${({ alignLoader }) => {
    if (alignLoader === 'top') {
      return `
        justify-items: start;
      `;
    }
    if (alignLoader === 'center') {
      return `
        justify-items: center;
      `;
    }
    return ``;
  }}

  ${({ zeroPadding }) => {
    if (zeroPadding) {
      return `
        padding: 0;
      `;
    }
    return ``;
  }}
`;

const LoadingOverlay = (props) => {
  const { alignLoader, loaderSize, zeroPadding } = props;

  const [isInfoVisible, setIsInfoVisible] = useState(false);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsInfoVisible(true);
    }, 8000);
    return () => clearTimeout(timeout);
  }, [setIsInfoVisible]);

  let info = false;
  if (isInfoVisible) {
    info = <span className="info">Just a second</span>;
  }

  return (
    <SC.Wrapper alignLoader={alignLoader} zeroPadding={zeroPadding}>
      <Loader size={loaderSize} />
      {info}
    </SC.Wrapper>
  );
};

LoadingOverlay.defaultProps = {
  zeroPadding: false,
};

LoadingOverlay.propTypes = {
  zeroPadding: PropTypes.bool,
  alignLoader: PropTypes.string.isRequired,
  loaderSize: PropTypes.string.isRequired,
};

export default LoadingOverlay;
