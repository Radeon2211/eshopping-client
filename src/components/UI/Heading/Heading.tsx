import React from 'react';
import * as SC from './Heading.sc';
import { PropsWithChildren } from '../../../shared/types/types';
import { ThemeLineHeight, ThemeSpacing } from '../../../styled/theme';

export interface HeadingProps {
  $variant: string;
  $align?: string;
  $mgBottom?: ThemeSpacing;
  $mgTop?: ThemeSpacing;
  $lineHeight?: ThemeLineHeight;
  $fontSize?: string;
  className?: string;
}

const Heading = React.memo((props: PropsWithChildren<HeadingProps>) => {
  return (
    <SC.Heading {...props} $align={props.$align || 'left'}>
      {props.children}
    </SC.Heading>
  );
});

export default Heading;

Heading.displayName = 'Heading';
