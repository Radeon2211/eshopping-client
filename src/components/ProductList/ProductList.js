import PropTypes from 'prop-types';
import { useLocation } from 'react-router-dom';
import styled from 'styled-components';
import ProductItem from './ProductItem/ProductItem';
import LoadingOverlay from '../UI/LoadingOverlay/LoadingOverlay';
import Heading from '../UI/Heading/Heading';
import { productPages } from '../../shared/constants';
import { getParamsWithoutPollution } from '../../shared/utility/utility';
import * as propTypes from '../../shared/propTypes';

const SC = {};
SC.Wrapper = styled.div`
  min-height: 4rem;
  position: relative;
  width: 100%;
`;

export default function ProductList({ products, isDataLoading, page }) {
  const { search } = useLocation();

  let loadingOverlay = null;
  if (isDataLoading) loadingOverlay = <LoadingOverlay alignLoader="start" />;

  let productList = null;
  if (products) {
    if (products.length <= 0) {
      let headingText = '';
      const correctQueryParams = getParamsWithoutPollution(search);
      if (correctQueryParams.p) delete correctQueryParams.p;
      const queryStringKeys = Object.keys(correctQueryParams);
      if (
        queryStringKeys.length >= 2 ||
        (!queryStringKeys.includes('name') && queryStringKeys.length === 1)
      ) {
        switch (page) {
          case productPages.ALL_PRODUCTS:
            headingText = `We didn't find any matching results. Try to search something else or change filters`;
            break;
          case productPages.MY_PRODUCTS:
          case productPages.USER_PRODUCTS:
            headingText = `We didn't find any matching results. Try to change filters`;
            break;
          default:
            headingText = 'Not found any results';
        }
      } else {
        switch (page) {
          case productPages.ALL_PRODUCTS:
            headingText = `We didn't find any matching results. Try to search something else`;
            break;
          case productPages.MY_PRODUCTS:
            headingText = `You don't have any offers published yet`;
            break;
          case productPages.USER_PRODUCTS:
            headingText = `This user doesn't have any offers published yet`;
            break;
          default:
            headingText = 'Not found any results';
        }
      }
      productList = (
        <Heading $variant="h4" $align="center" data-testid="ProductList-empty-list-info">
          {headingText}
        </Heading>
      );
    } else {
      productList = products.map((product) => <ProductItem key={product._id} data={product} />);
    }
  } else if (products === null) {
    productList = (
      <Heading $variant="h4" $align="center">
        There is a problem to fetch products
      </Heading>
    );
  }

  return (
    <SC.Wrapper data-testid="ProductList">
      {loadingOverlay}
      {productList}
    </SC.Wrapper>
  );
}

ProductList.propTypes = {
  // eslint-disable-next-line react/require-default-props
  products: PropTypes.arrayOf(PropTypes.shape(propTypes.fullProductItem)),
  isDataLoading: PropTypes.bool.isRequired,
  page: PropTypes.oneOf(Object.values(productPages)).isRequired,
};
