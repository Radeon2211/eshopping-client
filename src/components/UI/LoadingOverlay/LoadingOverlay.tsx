import { useState, useEffect } from 'react';
import Loader from '../Loader/Loader';
import PlainText from '../PlainText';
import * as SC from './LoadingOverlay.sc';

export interface LoadingOverlayProps {
  alignLoader?: string;
  loaderSize?: 'big' | 'medium' | 'small';
  disableText?: boolean;
}

export default function LoadingOverlay({
  alignLoader = 'center',
  loaderSize = 'medium',
  disableText = false,
}: LoadingOverlayProps) {
  const [isInfoVisible, setIsInfoVisible] = useState<boolean>(false);

  useEffect(() => {
    const timeout = setTimeout(() => {
      setIsInfoVisible(true);
    }, 8000);
    return () => clearTimeout(timeout);
  }, [setIsInfoVisible]);

  return (
    <SC.Wrapper $alignLoader={alignLoader} data-testid="LoadingOverlay">
      <Loader size={loaderSize} />
      {isInfoVisible && !disableText && (
        <PlainText $size="level2" $mgTop="level2" data-testid="LoadingOverlay-info">
          Just a second
        </PlainText>
      )}
    </SC.Wrapper>
  );
}
