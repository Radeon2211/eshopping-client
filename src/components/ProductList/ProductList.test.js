import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import { Router } from 'react-router-dom';
import theme from '../../styled/theme';
import ProductList from './ProductList';
import ProductItem from './ProductItem/ProductItem';
import Heading from '../UI/Heading/Heading';
import { checkProps } from '../../shared/utility';
import { pages } from '../../shared/constants';
import LoadingOverlay from '../UI/LoadingOverlay/LoadingOverlay';

const createHistory = (search = '') => ({
  listen: jest.fn(),
  createHref: jest.fn(),
  location: { search },
});

const defaultHistory = createHistory();

const setUp = (props, history = defaultHistory) => {
  return mount(
    <Router history={history}>
      <ThemeProvider theme={theme}>
        <ProductList {...props} />
      </ThemeProvider>
    </Router>,
  );
};

const createProps = (isDataLoading, products, page = pages.ALL_PRODUCTS) => ({
  isDataLoading,
  products,
  page,
});

describe('<ProductItem />', () => {
  describe('Check prop types', () => {
    it('Should NOT throw a warning', () => {
      const expectedProps = createProps(false);
      expect(checkProps(ProductList, expectedProps)).toBeUndefined();
    });
    it('Should throw a warning', () => {
      expect(checkProps(ProductList, {})).not.toBeNull();
    });
  });

  describe('Check if renders correctly', () => {
    it('Should render <LoadingOverlay />', () => {
      const props = createProps(true, null);
      const wrapper = setUp(props);
      expect(wrapper.find(LoadingOverlay)).toHaveLength(1);
    });
    it('Should NOT render <LoadingOverlay />', () => {
      const props = createProps(false, null);
      const wrapper = setUp(props);
      expect(wrapper.find(LoadingOverlay)).toHaveLength(0);
    });
    it('Should render one <ProductItem />', () => {
      const props = createProps(false, [{ _id: '123', name: 'testName', price: 2 }]);
      const wrapper = setUp(props);
      expect(wrapper.find(ProductItem)).toHaveLength(1);
    });
    it('Should render <Heading /> and NOT render <ProductItem />', () => {
      const props = createProps(false, []);
      const wrapper = setUp(props);
      expect(wrapper.find(ProductItem)).toHaveLength(0);
      expect(wrapper.find(Heading)).toHaveLength(1);
    });
    it('Should render info about changing results', () => {
      const props = createProps(false, []);
      const history = createHistory('?p=1&name=testName&condition=new');
      const wrapper = setUp(props, history);
      expect(wrapper.find(Heading).text()).toBe(
        `We didn't find any matching results. Try to search something else or change filters`,
      );
    });
    it('Should render info about trying something else', () => {
      const props = createProps(false, []);
      const history = createHistory();
      const wrapper = setUp(props, history);
      expect(wrapper.find(Heading).text()).toBe(
        `We didn't find any matching results. Try search something else`,
      );
    });
  });
});
