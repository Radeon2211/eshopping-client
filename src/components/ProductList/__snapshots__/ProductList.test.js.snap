// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem /> Check how renders Should render correct info for ALL_PRODUCTS with filters 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <h4
      class="sc-iBPRYJ kkqkio"
    >
      We didn't find any matching results. Try to search something else or change filters
    </h4>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render correct info for ALL_PRODUCTS without filters other than name 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <h4
      class="sc-iBPRYJ kkqkio"
    >
      We didn't find any matching results. Try to search something else
    </h4>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render correct info for MY_PRODUCTS with filters 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <h4
      class="sc-iBPRYJ kkqkio"
    >
      We didn't find any matching results. Try to change filters
    </h4>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render correct info for MY_PRODUCTS without filters other than name 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <h4
      class="sc-iBPRYJ kkqkio"
    >
      You don't have any offers published yet
    </h4>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render correct info for USER_PRODUCTS with filters 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <h4
      class="sc-iBPRYJ kkqkio"
    >
      We didn't find any matching results. Try to change filters
    </h4>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render correct info for USER_PRODUCTS without filters other than name 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <h4
      class="sc-iBPRYJ kkqkio"
    >
      This user doesn't have any offers published yet
    </h4>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render list with one item 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <a
      data-testid="ProductItem"
    >
      <div
        class="sc-bdfBwQ dXFlzG sc-gsTCUz vzdrx"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <div
          class="sc-bdfBwQ gEWXUp sc-hKgILt iGwkBA"
          data-testid="ProductThumbnail"
          direction="row"
          height="15"
          spacing=""
          width="13"
          wrap="nowrap"
        >
          <img
            alt="product1"
            data-testid="ProductThumbnail-img"
            src="no-photo.png"
          />
        </div>
        <div
          class="sc-bdfBwQ jPnCyv"
          direction="column"
          spacing=""
          width="100%"
          wrap="nowrap"
        >
          <span
            class="sc-dlfnbm LDMTW"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            product1
          </span>
          <span
            class="sc-dlfnbm iDQpJo"
            color=""
            data-testid="ProductItem-condition"
            display="inline-block"
            overflow="visible"
            size="1"
            spacing="0px"
            transform=""
          >
            <span
              class="sc-dlfnbm fMxJtk"
              color="#777"
              display="inline-block"
              overflow="visible"
              spacing="0px"
              transform=""
            >
              Condition:
            </span>
             New
          </span>
          <span
            class="sc-dlfnbm cQEZxx"
            color=""
            data-testid="ProductItem-price"
            display="inline-block"
            overflow="visible"
            size="5"
            spacing="0px"
            transform=""
          >
            $10.60
          </span>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render list with two items 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <a
      data-testid="ProductItem"
    >
      <div
        class="sc-bdfBwQ dXFlzG sc-gsTCUz vzdrx"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <div
          class="sc-bdfBwQ gEWXUp sc-hKgILt iGwkBA"
          data-testid="ProductThumbnail"
          direction="row"
          height="15"
          spacing=""
          width="13"
          wrap="nowrap"
        >
          <img
            alt="product1"
            data-testid="ProductThumbnail-img"
            src="no-photo.png"
          />
        </div>
        <div
          class="sc-bdfBwQ jPnCyv"
          direction="column"
          spacing=""
          width="100%"
          wrap="nowrap"
        >
          <span
            class="sc-dlfnbm LDMTW"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            product1
          </span>
          <span
            class="sc-dlfnbm iDQpJo"
            color=""
            data-testid="ProductItem-condition"
            display="inline-block"
            overflow="visible"
            size="1"
            spacing="0px"
            transform=""
          >
            <span
              class="sc-dlfnbm fMxJtk"
              color="#777"
              display="inline-block"
              overflow="visible"
              spacing="0px"
              transform=""
            >
              Condition:
            </span>
             New
          </span>
          <span
            class="sc-dlfnbm cQEZxx"
            color=""
            data-testid="ProductItem-price"
            display="inline-block"
            overflow="visible"
            size="5"
            spacing="0px"
            transform=""
          >
            $10.60
          </span>
        </div>
      </div>
    </a>
    <a
      data-testid="ProductItem"
    >
      <div
        class="sc-bdfBwQ dXFlzG sc-gsTCUz vzdrx"
        direction="row"
        spacing="3"
        width="100%"
        wrap="nowrap"
      >
        <div
          class="sc-bdfBwQ gEWXUp sc-hKgILt iGwkBA"
          data-testid="ProductThumbnail"
          direction="row"
          height="15"
          spacing=""
          width="13"
          wrap="nowrap"
        >
          <img
            alt="product2"
            data-testid="ProductThumbnail-img"
            src="no-photo.png"
          />
        </div>
        <div
          class="sc-bdfBwQ jPnCyv"
          direction="column"
          spacing=""
          width="100%"
          wrap="nowrap"
        >
          <span
            class="sc-dlfnbm LDMTW"
            color=""
            display="inline-block"
            overflow="visible"
            size="3"
            spacing="0px"
            transform=""
          >
            product2
          </span>
          <span
            class="sc-dlfnbm iDQpJo"
            color=""
            data-testid="ProductItem-condition"
            display="inline-block"
            overflow="visible"
            size="1"
            spacing="0px"
            transform=""
          >
            <span
              class="sc-dlfnbm fMxJtk"
              color="#777"
              display="inline-block"
              overflow="visible"
              spacing="0px"
              transform=""
            >
              Condition:
            </span>
             New
          </span>
          <span
            class="sc-dlfnbm cQEZxx"
            color=""
            data-testid="ProductItem-price"
            display="inline-block"
            overflow="visible"
            size="5"
            spacing="0px"
            transform=""
          >
            $299.98
          </span>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;

exports[`<ProductItem /> Check how renders Should render only <LoadingOverlay /> 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw kEEAfr"
    data-testid="ProductList"
  >
    <div
      class="sc-gKsewC jItbZz"
      data-testid="LoadingOverlay"
    >
      <div
        class="sc-jSgupP dKnIqn"
        data-testid="Loader"
      >
        <div />
        <div />
        <div />
        <div />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
